### PR TITLE
Add tests for OTEL_DOTNET_AUTO_*_ENABLED=false and OTEL_*_EXPORTER=none

### LINKED ISSUE #1630: SmokeTests for `OTEL_DOTNET_AUTO_*_ENABLED=false`
We should add a SmokeTests for `OTEL_DOTNET_AUTO_*_ENABLED=false` (can be as a separate PR). It should be very easy (based on https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1613).

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1629#pullrequestreview-1184131679_

### PR DESCRIPTION
**Pull request by birojnayak**
## Why

Fixes #1630

## What

Adds automated tests to increase code coverage/

## Tests

CI

## Checklist

- [x] ~~`CHANGELOG.md` is updated.~~
- [x] ~~Documentation is updated.~~
- [x] ~~New features are covered by tests.~~

### CODE REVIEW

File: test/IntegrationTests/Helpers/MockMetricsCollector.cs [hunk 1]
```diff
@@ -146,7 +146,7 @@ public void AssertExpectations(TimeSpan? timeout = null)
     internal void AssertEmpty(TimeSpan? timeout = null)
     {
         timeout ??= Timeout.NoExpectation;
-        if (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value))
+        if (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value) && metricsResource.Count > 0)
         {
             Assert.Fail($"Expected nothing, but got: {metricsResource}");
         }
```
Comment (pellared) [inline test/IntegrationTests/Helpers/MockMetricsCollector.cs:152]: We should check if all `_metricsSnapshots` are empty

I think it should be:

```suggestion
        while (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value))
        {
            if (metricsResource.Count > 0)
            {
                Assert.Fail($"Expected nothing, but got: {metricsResource}");
            }
        }
```

Current code may exit too early

File: test/IntegrationTests/SmokeTests.cs [hunk 2]
```diff
@@ -258,14 +258,16 @@ public void SubmitLogs()
         collector.AssertExpectations();
     }
 
-    [Fact]
+    [Theory]
+    [InlineData("OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS", "none")]
+    [InlineData("OTEL_DOTNET_AUTO_LOGS_ENABLED", "false")]
```
Comment (pellared) [inline test/IntegrationTests/SmokeTests.cs:263]: you can also add

```csharp
[InlineData("OTEL_LOGS_EXPORTER", "none")]
```

File: test/IntegrationTests/SmokeTests.cs [hunk 3]
```diff
@@ -274,22 +276,28 @@ public void LogsNoneInstrumentations()
 
 #endif
 
-    [Fact]
+    [Theory]
+    [InlineData("OTEL_DOTNET_AUTO_TRACES_ENABLED_INSTRUMENTATIONS", "none")]
+    [InlineData("OTEL_DOTNET_AUTO_TRACES_ENABLED", "false")]
```
Comment (pellared) [inline test/IntegrationTests/SmokeTests.cs:281]: you can also add

```csharp
[InlineData("OTEL_TRACES_EXPORTER", "none")]
```

File: test/IntegrationTests/SmokeTests.cs [hunk 4]
```diff
@@ -274,22 +276,28 @@ public void LogsNoneInstrumentations()
...
+        SetExporter(collector);
+        SetEnvironmentVariable(envVarName, envVarVal);
         RunTestApplication();
         collector.AssertEmpty();
     }
 
-    [Fact]
+    [Theory]
+    [InlineData("OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS", "none")]
+    [InlineData("OTEL_DOTNET_AUTO_METRICS_ENABLED", "false")]
```
Comment (pellared) [inline test/IntegrationTests/SmokeTests.cs:294]: you can also add

```csharp
[InlineData("OTEL_METRICS_EXPORTER", "none")]
```

**Review by pellared — CHANGES_REQUESTED** [review]

File: test/IntegrationTests/Helpers/MockMetricsCollector.cs (hunk 1 above)
Comment (birojnayak) [reply test/IntegrationTests/Helpers/MockMetricsCollector.cs:152]: @pellared the second condition will only executed if first is true...both (suggestion and what I have written) is same.. am I missing something ?

Comment (pellared) [reply test/IntegrationTests/Helpers/MockMetricsCollector.cs:152]: my suggestion has `while` in the first condition to make sure to read all `_metricsSnapshots` and not just the first one

File: test/IntegrationTests/Helpers/MockMetricsCollector.cs [hunk 5]
```diff
@@ -148,7 +148,10 @@ internal void AssertEmpty(TimeSpan? timeout = null)
         timeout ??= Timeout.NoExpectation;
         if (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value))
```
Comment (pellared) [inline test/IntegrationTests/Helpers/MockMetricsCollector.cs:149]:
```suggestion
        while (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value))
```

File: test/IntegrationTests/Helpers/MockMetricsCollector.cs (hunk 1 above)
Comment (pellared) [reply test/IntegrationTests/Helpers/MockMetricsCollector.cs]: See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1693#discussion_r1036221654

**Review by pellared — APPROVED** [review]

**Review by Kielek — APPROVED** [review]
